File: backend/app/features/categories/schemas/category.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional


class CategoryCreate(BaseModel):
    name: str
    created_by: Optional[int] = None
    updated_by: Optional[int] = None


class CategoryRead(BaseModel):
    id: int
    name: str
    created_at: Optional[datetime] = None
    created_by: Optional[int] = None
    updated_at: Optional[datetime] = None
    updated_by: Optional[int] = None
    is_deleted: bool = False
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/features/categories/types/category.ts
export interface Category {
  id: number;
  name: string;
  created_at?: string;
  created_by?: number | null;
  updated_at?: string;
  updated_by?: number | null;
  is_deleted: boolean;
}

export interface CategoryCreate {
  name: string;
  created_by?: number | null;
  updated_by?: number | null;
}

